From ba0fe79716d869195c1a16b0a8c848964427e59e Mon Sep 17 00:00:00 2001
From: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
Date: Fri, 3 Aug 2018 15:18:05 +0200
Subject: [PATCH] PMCPESYS-68: thermal: ltq: Register sensor with index 0 as
 well

'#thermal-sensor-cells=<0>' uses index 0 for the first sensor and it
should be also considered. In probe() index sensors from 1 to allocate
all sensors properly.

Signed-off-by: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
---
 drivers/thermal/ltq_thermal.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/thermal/ltq_thermal.c b/drivers/thermal/ltq_thermal.c
index 2438926c6545..f662ea6c82eb 100644
--- a/drivers/thermal/ltq_thermal.c
+++ b/drivers/thermal/ltq_thermal.c
@@ -260,11 +260,9 @@ static int ltq_thermal_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->chiptop);
 	}
 
-	if (device_property_read_u32(&pdev->dev, "#thermal-sensor-cells",
-				     &priv->count)) {
-		/* register single cell if #thermal-sensor-cells is missing */
-		priv->count = 0;
-	}
+	device_property_read_u32(&pdev->dev, "#thermal-sensor-cells",
+				 &priv->count);
+	priv->count++;
 
 	priv->sensors = devm_kzalloc(&pdev->dev, priv->count *
 				     sizeof(struct ltq_thermal_sensor),
